src/components/common/ShareButton/ShareButton.tsx

```diff
@@ -0,0 +1,37 @@
+import Image from 'next/image'
+import React from 'react'
+import { AiOutlineCopy } from 'react-icons/ai'
+import { toast, ToastContainer } from 'react-toastify'
+import { ShareButtonProps } from './ShareButton.type'
+import 'react-toastify/dist/ReactToastify.css'
+
+const ShareButton: React.FC<ShareButtonProps> = () => {
+  const handleCopy = async () => {
+    try {
+      await navigator.clipboard.writeText(location.href)
+      toast.success('URLをコピーしました')
+    } catch (error) {
+      alert('URLをコピーできませんでした')
+    }
+  }
+
+  const handleShareLine = () => {
+    // Lineに共有するURLを作成
+    const lineShareUrl = `https://social-plugins.line.me/lineit/share?url=${location.href}`
+    window.open(lineShareUrl, '_blank') // 新しいタブで開く
+  }
+
+  return (
+    <div className='flex items-center'>
+      <button className='btn btn-ghost' onClick={handleCopy}>
+        <AiOutlineCopy size='25' />
+      </button>
+      <div onClick={handleShareLine}>
+        <Image src='/LINE_Brand_icon.png' width={36} height={36} alt='logo' />
+      </div>
```

Comment:
押す要素であればbuttonで定義しよう

```suggestion
      <button onClick={handleShareLine}>
        <Image src='/LINE_Brand_icon.png' width={36} height={36} alt='logo' />
      </div>
```

Reply:
[4731047](https://github.com/NUTFes/chosei-chan/pull/32/commits/4731047719e73480e6d68f2c61400b1e7eea208f)